### ISSUE TITLE
Support Suppressing Phone Call Notifications When Screen On

### ISSUE THREAD

**Issue by srwareham** [Freeyourgadget/Gadgetbridge#1450]
#### Before requesting a new feature, please confirm the following:
- [x ] I have read the [wiki](https://github.com/Freeyourgadget/Gadgetbridge/wiki), and I didn't find a solution to my problem / an answer to my question.
- [ x] I have searched the [issues](https://github.com/Freeyourgadget/Gadgetbridge/issues), and I didn't find a solution to my problem / an answer to my question.
- [ x] If you upload an image or other content, please make sure you have read and understood the [github policies and terms of services](https://help.github.com/articles/github-terms-of-service/#1-responsibility-for-user-generated-content)

#### Your issue is:
SMS notifications contain the option to show notifications only "When screen is off." Phone calls, however, only have the options "Never" and "Always." It would be greatly helpful to support "When screen is off" for phone calls.

#### Your wearable device is:

Amazfit Bip firmware version V1.1.2.05

#### Your android version is:

Android 9. Pixel 3. 

#### Your Gadgetbridge version is:

0.32.3

**Comment by srwareham** [comment]
To any devs, are there any technological limitations why there's a divergence in the settings between phone calls and SMS? If not, I'm happy to do the work myself for putting together a PR. That said, any pointers in the right direction would be greatly appreciated!

Thanks for the great work! No chance I'd have ever bought a smartwatch without you all.

**Comment by ashimokawa** [comment]
@srwareham 
Yes, there is a difference - most phones turn on the screen when ringing - it would be a race condition. Any ideas on how to solve that problem?

And thanks - without being willing to start that I would never have bought my first smartwatch ;)

**Comment by srwareham** [comment]
@ashimokawa 
I can try some tests, but wouldn't the same condition exist for SMS (i.e., the screen becomes active to show the message)? Looking into the implementation of [SMSReceiver.java](https://github.com/Freeyourgadget/Gadgetbridge/blob/master/app/src/main/java/nodomain/freeyourgadget/gadgetbridge/externalevents/SMSReceiver.java), couldn't the same methodology simply be ported over? Or have you already tried and the screen is considered interactive when the phone call answer screen appears? Also noticed that the `isScreenOn()` method used is [deprecated](https://developer.android.com/reference/android/os/PowerManager.html#isScreenOn()) as of API 20, if at all helpful.

I'm not at all experienced with the Android data model etc., but does [NotificationListener.java](https://github.com/Freeyourgadget/Gadgetbridge/blob/master/app/src/main/java/nodomain/freeyourgadget/gadgetbridge/externalevents/NotificationListener.java) perhaps capture the Intent before the screen is woken? If so, maybe it could create a state variable to be read by [PhoneCallReceiver.java](https://github.com/Freeyourgadget/Gadgetbridge/blob/master/app/src/main/java/nodomain/freeyourgadget/gadgetbridge/externalevents/PhoneCallReceiver.java)?

If the above fail, maybe there's a way to access the phone call activity and evaluate what the scheduled `onResume()` action is? That would be to say if any app (besides the home screen?) is to take over next, we would know the screen was on. I also assume there could be a somewhat less tidy implementation where one could simply check if the user has authenticated their device login (i.e., entered a password or used fingerprint unlock). This would of course only work for users with authentication enabled on their device.
